File: Dataset/Leetcode/train/5/160.py
 class Solution:
    def XXX(self, s: str) -> str:
        temp,L =[s],0
        for x in range(len(s)):
            for y in range(L,len(s)-x+1):
                Str = s[x:x+y]
                if Str == Str[::-1] and y>=L:
                    L=y
                    temp.append(Str)
        return temp[-1]

File: Dataset/Leetcode/test/8/160.cpp
class Solution {
public:
    int XXX(string str) {
        int i = 0, n = str.size(),sign = 1,ans = 0;
        //把i移动非空格位
        for(;i < n && isspace(str[i]);i++);
        //判断符号
        if(str[i] == '+' || str[i] == '-'){
            sign *= (str[i++] == '+' ? 1:-1);
        }
        
        while(str[i] >= '0' && str[i] <= '9'){
            //std::cout << ans;
            if(ans > INT_MAX/10 || (ans == INT_MAX/10 && str[i]-'0'>7)){
                if (sign == 1) return INT_MAX;
                else return INT_MIN;
            }
            ans = ans * 10 + (str[i++]-'0');
        }
        return ans * sign;
        
    }
};

